### ISSUE TITLE
Regression in how includes from .h are handled in the corresponding .cpp

### ISSUE THREAD

**Issue by elrinor** [cpplint/cpplint#251]
test.h:
```
#pragma once

#include <string>
```

test.cpp:
```
#include "test.h"

#define TEST(S)

TEST(string)

void f() {}
```

cpplint command:
```
python3 ./cpplint.py --filter=-legal/copyright,-build/include_subdir test.h test.cpp
```

Output before e3482c53590e9ff61280a0cb5cd61fc692570fa6:
```
Done processing test.cpp
Done processing test.h
```

Output after e3482c53590e9ff61280a0cb5cd61fc692570fa6:
```
test.cpp:5:  Add #include <string> for string  [build/include_what_you_use] [4]
Done processing test.cpp
Done processing test.h
Total errors found: 1
```

Looks like a regression to me, found this after updating to the latest develop and discovering that I now have a bunch of cpplint errors in my project.

**Comment by elrinor** [comment]
@aaronliu0130 can you take a look?

Looking at the change, it seems intentional, but I wonder what was the reasoning behind it? 

Removing the header rule just leads to more includes, which are 100% redundant.

**Comment by aaronliu0130** [comment]
I intended this to be in the commit description, but I forgot that and now it's a comment to the commit.

> This fixes https://github.com/google/styleguide/issues/757

TL;DR: That's what include what you use means, and that's how the style guide wants it.

**Comment by elrinor** [comment]
OK it's even approved by Titus, thanks.
